frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.tsx

```diff
@@ -166,40 +201,63 @@ export function MultiAutocomplete({
     <span />
   );
 
+  const CustomItemComponent = useCallback(
+    (props: SelectItemProps) => {
+      const customLabel =
+        props.value !== undefined && renderValue?.(props.value as ValueType);
+      return <ItemWrapper {...props} label={customLabel ?? props.label} />;
+    },
+    [renderValue],
+  );
+
   return (
     <MultiSelect
       {...props}
+      // @ts-expect-error: Mantine's types expects a string value,
+      // but does not depend on it being a string
       data={items}
```

Comment:
Mantine's type forces the value to be `string` but it can be anything (as long as it is comparable using `===`).